webapp/src/app/shared/phaser/entities/registry/entity-registry.service.ts

```diff
@@ -1,45 +1,43 @@
-import {DefaultEntity} from './default-entity';
-import {CCEntity} from '../cc-entity';
-import {Prop} from './prop';
-import {ScalableProp} from './scalable-prop';
-import {ItemDestruct} from './item-destruct';
-import {NPC} from './npc';
-import {EventTrigger} from './event-trigger';
-import {Injectable} from '@angular/core';
+import { DefaultEntity } from './default-entity';
+import { Prop } from './prop';
+import { ScalableProp } from './scalable-prop';
+import { ItemDestruct } from './item-destruct';
+import { NPC } from './npc';
+import { EventTrigger } from './event-trigger';
+import { Injectable } from '@angular/core';
 import { Destructible } from './destructible';
+import { CCEntity } from '../cc-entity';
 
 @Injectable()
 export class EntityRegistryService {
-	private entities: { [type: string]: any } = {};
-	private defaultEntity: any;
+	private entities: { [type: string]: typeof CCEntity } = {};
+	private defaultEntity = DefaultEntity;
 	
 	constructor() {
-		this.setDefaultEntity(DefaultEntity);
 		this.register('Prop', Prop);
 		this.register('ScalableProp', ScalableProp);
 		this.register('ItemDestruct', ItemDestruct);
 		this.register('Destructible', Destructible);
 		this.register('NPC', NPC);
+		this.register('NPC', NPC);
 		this.register('EventTrigger', EventTrigger);
 	}
 	
-	private setDefaultEntity(entity: any) {
-		this.defaultEntity = entity;
-	}
-	
-	private register(type: string, entity: any) {
+	private register(type: string, entity: typeof CCEntity) {
 		this.entities[type] = entity;
 	}
 	
-	public getDefaultEntity(): new (...args: any[]) => CCEntity {
+	public getDefaultEntity(): typeof DefaultEntity {
 		return this.defaultEntity;
 	}
 	
 	public getAll() {
 		return this.entities;
 	}
 	
-	public getEntity(type: string): new (...args: any[]) => CCEntity {
-		return this.entities[type] || this.defaultEntity;
+	// typed as DefaultEntity so constructor can be used with parameter checking.
+	// CCEntity is abstract and doesn't allow using the constructor
+	public getEntity(type: string): typeof DefaultEntity {
+		return (this.entities[type] || this.defaultEntity) as any;
```

Comment:
Why not use `DefaultEntity` directly? 
```suggestion
		return (this.entities[type] ?? DefaultEntity) as typeof DefaultEntity
```
(I don't care of one does `??` or `||` in this case, just a suggestion)